### PR TITLE
Add Delay Load Hook to make compatible with electron >= 4.0.0

### PR DESCRIPTION
**Pull request by rogeriorc**
Add win_delay_load_hook.cc and Linker Flag /DELAYLOAD:NODE.EXE to windows build

Added CMAKE_JS_SRC cmake variable to be added to sources 

**Related issues:**

* electron/electron#17326
* electron/electron#17462

### CODE REVIEW

**Comment by rogeriorc** [comment]
@unbornchikken what do you think about this approach?

**Comment by Djiit** [comment]
Can we help on this ? We'd love to see this PR merged.

**Comment by anoadragon453** [comment]
Still waiting on this for compatibility with Electron ABI >v69.

**Comment by unbornchikken** [comment]
Released as of v5.2.0.

File: lib/es6/cMake.js [hunk 1]
```diff
@@ -158,6 +158,16 @@ CMake.prototype.getConfigureCommand = async(function* () {
     // Includes:
     D.push({"CMAKE_JS_INC": incPaths.join(";")});
 
+    // Sources:
+    let srcPaths = [];
+    if (environment.isWin) {
```
Comment (BotellaA) [inline lib/es6/cMake.js]: Should it be restricted to runtime == electron only?

File: lib/cpp/win_delay_load_hook.cc [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#ifdef _MSC_VER
```
Comment (BotellaA) [inline lib/cpp/win_delay_load_hook.cc]: I found this file that seems to be more complete but I do not really understand what it does 😛

**Pull request merged**